### PR TITLE
feat(Bar): Add new prop striped

### PR DESCRIPTION
**Pull request by stefanbuck**
Adding a new property `fillMode` to apply a different background texture. [Demo](https://brave-keller-55c6a8.netlify.com/docs/packages/axiom-charts/bars/)

### solid (default)
screen shot 2018-05-08 at 10 42 59

### semstriped
screen shot 2018-05-08 at 10 42 04

### striped
screen shot 2018-05-08 at 10 42 13

### CODE REVIEW

File: packages/axiom-charts/src/Bar/Bar.js [hunk 1]
```diff
@@ -55,13 +55,16 @@ export default class Bar extends Component {
     showLabel: PropTypes.bool,
     /** Explicit thickness of the Bar */
     size: PropTypes.string,
+    /** When true the background will be striped */
+    stripedBackground: PropTypes.bool,
```
Comment (stefanbuck) [inline packages/axiom-charts/src/Bar/Bar.js]: In the PR description I was writing `texture` background instead of `striped` which I think sounds better and fits if we decide to use dotted backgrounds in the future.

File: packages/axiom-charts/src/Bar/Bar.css [hunk 2]
```diff
@@ -240,3 +240,13 @@
...
   }
 }
+
+.ax-bars__bar-rect--striped {
+  background-image: repeating-linear-gradient(
+  -45deg,
+  transparent,
+  transparent 10px,
+  rgba(255, 255, 255, .3) 10px,
+  rgba(255, 255, 255, .3) 20px
```
Comment (HHogg) [inline packages/axiom-charts/src/Bar/Bar.css]: Do these colour values need to reference a swatch colour and the size of the bars to the grid system?

File: packages/axiom-charts/src/Bar/Bar.css [hunk 3]
```diff
@@ -240,3 +240,14 @@
     padding-top: var(--cmp-chart-label-margin);
   }
 }
+
+.ax-bars__bar-rect--striped {
+  background-image:
+    repeating-linear-gradient(
+      -45deg,
+      transparent,
```
Comment (HHogg) [inline packages/axiom-charts/src/Bar/Bar.css]: Should this be a translucent colour of the bars colour to handle the exceeded difference situation? (I guess when the difference is less than value the translucent colour will appear transparent against the bar)

File: packages/axiom-charts/src/Bar/Bar.css [hunk 4]
```diff
@@ -240,3 +240,14 @@
...
+      transparent 0.7rem,
+      rgba(255, 255, 255, 0.3) 0.7rem,
+      rgba(255, 255, 255, 0.3) 1.4rem
```
Comment (HHogg) [inline packages/axiom-charts/src/Bar/Bar.css]: Do these values need to correspond to the swatch colours and the grid system?

File: packages/axiom-charts/src/Bar/Bar.js (hunk 1 above)
Comment (HHogg) [reply packages/axiom-charts/src/Bar/Bar.js]: `striped` Sounds good. Do you think as we just use `color` (not `backgroundColor`) for the background colour configuration, we should just go with `striped` rather than `stripedBackground` for consistency?

Comment (stefanbuck) [reply packages/axiom-charts/src/Bar/Bar.js]: `striped` works for me

File: packages/axiom-charts/src/Bar/Bar.css (hunk 4 above)
Comment (stefanbuck) [reply packages/axiom-charts/src/Bar/Bar.css]: As I mentioned before, using 30% white was just the quickest way to achieve this. If we want to be explicit I need a list of all corresponding colour values.

Comment (HHogg) [reply packages/axiom-charts/src/Bar/Bar.css]: Yeah, the transparent white I agree is the way to go, but I expect the opacity from the designs  comes from one of the predefined ones in [here](https://github.com/BrandwatchLtd/axiom/blob/master/packages/axiom-materials/src/opacities.css).

File: packages/axiom-charts/src/Bar/Bar.css (hunk 3 above)
Comment (stefanbuck) [reply packages/axiom-charts/src/Bar/Bar.css]: Ignore my comment above, I mixed up things, sorry.

In the exceeded cases there will be a bar underneath as well which has also a background colour.

Comment (HHogg) [reply packages/axiom-charts/src/Bar/Bar.css]: No worries :) You should be able to avoid defining a colour for every gradient by using the `currentColor` trick.

File: packages/stylelint-config-axiom/src/stylelintrc.json [hunk 5]
```diff
@@ -53,7 +53,7 @@
     "indentation": 2,
     "length-zero-no-unit": true,
     "max-empty-lines": 1,
-    "max-line-length": 110,
+    "max-line-length": 120,
```
Comment (stefanbuck) [inline packages/stylelint-config-axiom/src/stylelintrc.json]: The previous value was conflicting with changes I made in `Bar.css`. I think it's suitable to tweak the rule rather than updating every other line which makes mutli cursor editing way more easier.

File: packages/axiom-charts/src/Bar/Bar.js [hunk 6]
```diff
@@ -62,6 +72,7 @@ export default class Bar extends Component {
     isHidden: false,
     labelStrong: false,
     minSize: '1rem',
+    fillMode: FILL_MODE_SOLID,
```
Comment (stefanbuck) [inline packages/axiom-charts/src/Bar/Bar.js]: @HHogg I sticked with the `fillMode` property to have full control over the bar. For example, in my design I want to have a centered tooltip on the solid bar, not both bars. 

screen shot 2018-05-04 at 11 06 35

File: packages/axiom-charts/src/Bar/Bar.css [hunk 7]
```diff
@@ -28,166 +28,362 @@
...
+        currentColor calc(var(--cmp-grid-gutter-size-tiny) * 2)
+      );
+  }
+
+  & .ax-bars__bar-striped--transparent {
+    opacity: 1;
+  }
+
+  &.ax-bars__bar-rect--tiny-clanger .ax-bars__bar-filled,
+  &.ax-bars__bar-rect--tiny-clanger .ax-bars__bar-striped { color: var(--color-product-tiny-clanger); }
```
Comment (HHogg) [inline packages/axiom-charts/src/Bar/Bar.css]: I think you could probably just go with 

```css
.ax-bars__bar-rect--tiny-clanger { color: var(--color-product-tiny-clanger); }
```

As `.ax-bars__bar-filled` and `.ax-bars__bar-striped` are descendants the `currentColor` will work from the inherited value. The same with the `:hover` and `:active` selectors below. 

This will likely drop those lines lengths too, so the linting rule could be kept at it's original value :)

File: packages/axiom-charts/src/Bar/Bar.js (hunk 6 above)
Comment (HHogg) [reply packages/axiom-charts/src/Bar/Bar.js]: Ok that makes sense 👍

File: packages/axiom-charts/src/Bar/Bar.css [hunk 8]
```diff
@@ -28,166 +28,362 @@
...
+    top: 0;
+    bottom: 0;
+    opacity: 0.3;
+    background:
+      repeating-linear-gradient(
+        -45deg,
+        rgb(var(--rgb-ui-white)),
+        rgb(var(--rgb-ui-white)) var(--cmp-grid-gutter-size-tiny),
+        currentColor var(--cmp-grid-gutter-size-tiny),
+        currentColor calc(var(--cmp-grid-gutter-size-tiny) * 2)
```
Comment (HHogg) [inline packages/axiom-charts/src/Bar/Bar.css]: Using the `--cmp-grid-gutter` variables won't preprocess because those variables belong to the Grid component. Only variables from the materials package should be used across other files. 

**tldr;** Instead of `--cmp-grid-gutter-size-tiny` use `--spacing-grid-size--x2`

File: packages/axiom-charts/src/Bar/Bar.css [hunk 9]
```diff
@@ -28,166 +28,362 @@
...
+      );
+  }
+
+  & .ax-bars__bar-striped--transparent {
+    opacity: 1;
```
Comment (HHogg) [inline packages/axiom-charts/src/Bar/Bar.css]: I'm not 100% sure this is correct. Looking at the standalone striped designs, they still remain at 0.3

screen shot 2018-05-04 at 10 50 36

File: packages/axiom-charts/src/Bar/Bar.css (hunk 7 above)
Comment (stefanbuck) [reply packages/axiom-charts/src/Bar/Bar.css]: Done see fixup commit 71703cb8

File: packages/axiom-charts/src/Bar/Bar.css (hunk 9 above)
Comment (stefanbuck) [reply packages/axiom-charts/src/Bar/Bar.css]: Fixed with fixup 47b5cf42

**Comment by stefanbuck** [comment]
Hi @HHogg, I addressed each of your comments in a separate fixup commit. The last fixup renames some classNames so that those are inline with the props names.
